README.md

```diff
@@ -16,370 +14,85 @@ It's aimed mainly to testing WildFly application server.
-import org.wildfly.extras.sunstone.api.Node;
-
-import com.squareup.okhttp.OkHttpClient;
-import com.squareup.okhttp.Request;
-
-/**
- * Sunstone quick-start. It creates cloud provider, starts a single node in it and
- * makes HTTP request (port 8080) against the node.
- */
-public class App {
-
-	public static void main(String[] args) throws IOException {
-		CloudProperties.getInstance().load("my-cloud.properties");
-		try (CloudProvider cloudProvider = CloudProvider.create("my-provider")) {
-			try (Node node = cloudProvider.createNode("wildfly-node")) {
-				final URL url = new URL("http", node.getPublicAddress(), 8080, "");
-				final OkHttpClient client = new OkHttpClient();
-				final Request nodeNameRequest = new Request.Builder().url(url).build();
-				System.out.println("Response from " + url);
-				System.out.println(client.newCall(nodeNameRequest).execute().body().string());
-			}
-		}
-	}
+@Setup(SetupTest.StaticClassTask.class)
+public class SetupTest {
+
+    @Test
+    public void test() {
+        // test logic
+    }
+
+    static class StaticClassTask extends AbstractSetupTask {
+        @Override
+        public void setup() throws Exception {
+            // configure WildFly or cloud resources
+        }
+
+        @Override
+        public void teardown() throws Exception {
+            // tear down the configuration
+        }
+    }
 }
 ```
 
-## Clouds support
+###### WildFly deployment
 
-Available:
-* [Docker](docker-README.md)
-* [Microsoft Azure](azure-README.md) and, as a technical preview, [Microsoft Azure (ARM)](azure-arm-README.md)
-* [Amazon EC2](ec2-README.md)
-* [OpenStack](openstack-README.md)
-* [Bare Metal](baremetal-README.md) (not really a cloud provider per se, but useful anyway)
+WildFly deploy operation is specific to the type of resources WildFly is running on. The archive that you want to be deployed, you must use `@Deployment` annotation on a static method returning the archive. Supported return types:
+- File (local)
+- Path (local)
+- Archive (Shrinkwrap)
+- InputStream
 
-## WildFly/EAP support
+The method must be annotated by cloud-specific annotation that identifies the cloud resource. The annotations come from cloud specific Sunstone modules. For more examples see:
+* [Azure deploy](Azure-README.md#wildfly-deployment)
+* [AWS deploy](AWS-README.md#wildfly-deployment)
 
-* [WildFlyNode wrapper](wildfly-README.md)
-* [Arquillian extension](arquillian-README.md)
+###### Injection
+
+Sunstone support injection of several resources, i.e. hostname of resources (Azure Web app, AWS EC2 instance, ... ). The logic is that if you want to inject a particular field od particular type, you need to identify the cloud resource. I.e. you want to inject a hostname. The type must be `Hostname`. You also need to annotate the field with cloud-specific annotation that identify such resources. The annotations come from cloud specific Sunstone modules. For more examples see:
```

Comment:
"Sunstone supports"
And the previous point still stands - I think it would be better to have the example, starting from "I.e." on a new line.